src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -3264,8 +3269,6 @@ await thisParam.StartResourceSetAsync(resourceSetParam)
         private async Task VerifyCanWriteStartDeltaResourceSetAsync(bool synchronousCall, ODataDeltaResourceSet deltaResourceSet)
         {
             ExceptionUtils.CheckArgumentNotNull(deltaResourceSet, "deltaResourceSet");
-
-            this.VerifyWritingDelta();

```

Comment:
Curious why was this removed

Reply:
Previously, this checked to see whether the writer was created with isDelta=true.

Such a check was unnecessary, and was legacy from when we had different writers under the covers for delta versus non delta collections.

It should be valid to start writing a delta resource set regardless of how the writer was created. In fact, calling writeStartDeltaResourceSet is what sets the writingDelta to true. The only thing writingDelta should validate is whether we are currently in the context of writing a delta collection (in which case we can write a deletedResource or added/deleted link) or we are not (in which case we can't).

This is particularly important for enabling writing a nested delta set within a single resource (for example, for a patch request/response) since there is no way to specify writingDelta when creating a reader/writer for a single resource.